lib/providers/token.js

```diff
@@ -176,10 +176,26 @@ class TokenProvider {
         };
 
         return this.data;
+      }).catch((err) => {
+        // Only consider StatusCodeError to be a failing condition. Anything else could be network
+        // related or something else that we want to retry. StatusCodeError comes from request-promise which Vaulted
+        // uses to hit Vault.
+        if (err.name === 'StatusCodeError') {
+          throw err;
```

Comment:
Do we want to add some logging here since we're eating anything that's not a `StatusCodeError`? There may be other cases we want to throw on we don't know about yet.